src/pj_param.c

```diff
@@ -1,20 +1,78 @@
 /* put parameters in linked list and retrieve */
-#include <projects.h>
 #include <stdio.h>
+#include <ctype.h>
 #include <string.h>
+#include "projects.h"
 
 /* create parameter list entry */
 paralist *pj_mkparam(char *str) {
-	paralist *newitem;
-
-	if((newitem = (paralist *)pj_malloc(sizeof(paralist) + strlen(str))) != NULL) {
-		newitem->used = 0;
-		newitem->next = 0;
-		if (*str == '+')
-			++str;
-		(void)strcpy(newitem->param, str);
-	}
-	return newitem;
+    paralist *newitem;
+
+    if((newitem = (paralist *)pj_malloc(sizeof(paralist) + strlen(str))) != NULL) {
+        newitem->used = 0;
+        newitem->next = 0;
+        if (*str == '+')
+            ++str;
+        (void)strcpy(newitem->param, str);
+    }
+    return newitem;
+}
+
+
+/* As pj_mkparam, but payload is terminated by whitespace, rather than 0 */
+paralist *pj_mkparam_ws (char *str) {
+    paralist *newitem;
+    size_t len = 0;
+
+    if (0==str)
+        return 0;
+
+    /* Find start and length of string */
+    while (isspace (*str))
+        str++;
+    while ((!isspace(str[len])) && 0!=str[len])
+        len++;
+    if (*str == '+') {
+        str++;
+        len--;
+    }
+
+    /* Use calloc to automagically 0-terminate the strncpy */
+    newitem = (paralist *) pj_calloc (1, sizeof(paralist) + len);
+    if (0==newitem)
+        return 0;
+    strncpy(newitem->param, str, len);
```

Comment:
So there is pj_strlcpy and yet here you directly use strncpy?  What is the proj4 policy on the strcpy family of functions?

Reply:
I don't think there is any policy, but the comment by @aaronpuchert above, we should probably eliminate all calls to `strncpy` in order to be compliant with the explicit policy of being straight C89.